packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts

```diff
@@ -0,0 +1,366 @@
+  readonly version?: number,
+}
+
+/**
+ * Represents Contributor Insight Rule
+ */
+export interface IInsightRule extends IResource {
+
+  /**
+   * Name of this insight rule
+   *
+   * @attribute
+   */
+  readonly insightRuleRuleName: string;
+
+  /**
+   * Get a metric for this Insight Rule
+   *
+   * @param metricName The name of the metric to get
+   * @param metricOptions Additional metric options to apply
+   */
+  metric(metricName: string, metricOptions?: MathExpressionOptions): IMetric;
+}
+
+/**
+ * The base class for an Insight Rule
+ */
+abstract class InsightRuleBase extends Resource implements IInsightRule {
+  /**
+   * The name of this InsightRule
+   *
+   * @attribute
+   */
+  public abstract readonly insightRuleRuleName: string;
+
+  public metric(metricName: string, metricOptions: MathExpressionOptions = {}): IMetric {
+    return new MathExpression({
+      ...metricOptions,
+      expression: `INSIGHT_RULE_METRIC('${this.insightRuleRuleName}', '${metricName}')`,
+    });
+  }
+}
+
+/**
+ * A Contributor Insight Rule Construct
+ */
+export class InsightRule extends InsightRuleBase {
+
+  /**
+  * Import an existing Insight Rule from the rule name
+  *
+  * @param scope The parent creating construct (usually `this`)
+  * @param id The Construct's name
+  * @param insightRuleName The name of the Insight Rule
+  */
+  static fromInsightRuleName(scope: Construct, id: string, insightRuleName: string): IInsightRule {
+    class Import extends InsightRuleBase {
+      public readonly insightRuleRuleName = insightRuleName;
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * Name of this Insight Rule
+   *
+   * @attribute
+   */
+  public readonly insightRuleRuleName: string;
+
+  /**
+   * ARN of this Insight Rule (i.e. arn:aws:cloudwatch:<region>:<account-id>:insight-rule/Foo)
+   *
+   * @attribute
+   */
+  public readonly insightRuleArn: string;
+
+  constructor(scope: Construct, id: string, props: InsightRuleProps) {
+    super(scope, id, {
+      physicalName: props.insightRuleName,
+    });
+
+    const ruleState = props.enabled === false ? 'DISABLED' : 'ENABLED';
+
+    const {
+      aggregateOn,
+      filters,
+      logGroupNames,
+      logFormat,
+      keys,
+      schemaName,
+      sumValue,
+      version,
+    } = props;
+
+    if (filters && filters.length > 4) {
+      throw new Error('Only 4 Filters can be applied to an Insight Rule');
+    }
+
+    if (aggregateOn === AggregateOptions.SUM && !sumValue) {
+      throw new Error('`sumValue` must be specified if using Aggregate Option SUM');
+    }
+
+    const ruleBody = JSON.stringify({
+      AggregateOn: aggregateOn || AggregateOptions.COUNT,
+      Contribution: {
+        Keys: keys,
+        Filters: (filters || []).map((f: Filter) => ({
+          Match: f.match,
+          [f.operator]: f.target,
+        })),
+        ...(sumValue && { ValueOf: sumValue }),
+      },
+      Schema: {
+        Name: schemaName ?? 'CloudWatchLogRule',
```

Comment:
is this a reasonable default name? what if someone creates multiple Insight Rules, that will then have the same schema name?

Reply:
The value of Schema for a rule that analyzes CloudWatch Logs data must always be
`{"Name": "CloudWatchLogRule", "Version": 1}`
https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContributorInsights-RuleSyntax.html
I made it customizable anyways in case that changes or if someone knows something the docs aren't letting on

Reply:
given this information, please disregard my last comment re: versioning. We should _hardcode_ the `{"Name": "CloudWatchLogRule", "Version": 1}` and not make it configurable, so people will not shoot themselves in the foot. if that is the only valid value, then there's no reason to expose it. doing so is future-proofing :)